src/event.rs

```diff
@@ -389,8 +389,11 @@ pub enum WindowEvent {
     /// Applications might wish to react to this to change the theme of the content of the window
     /// when the system changes the window theme.
     ///
+    /// This only reports a change if the window theme was not overridden by the application.
+    ///
     /// ## Platform-specific
     ///
+    /// - **Web:** Currently reports a change whether the window theme was overridden or not.
```

Comment:
I am unsure if this is the desired behaviour here?

I think the Windows implementation checks the value set by the user, and doesn't emit an event in that case, and it seems like the Web backend doesn't do such a check? @daxpedda?

Reply:
> I think the Windows implementation checks the value set by the user, and doesn't emit an event in that case, and it seems like the Web backend doesn't do such a check?

Web doesn't have a way to set the theme, ergo `Window::set_theme()` and `WindowAttributes::with_theme()` are no-op on Web. So I think your proposed changes are fine except the platform-specific addition to Web, as there is no way to overwrite the theme on Web.

> In reality, the `ThemeChanged` event is really an application-wide thing, even on Windows (the [`WM_SETTINGCHANGE`](https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-settingchange) event is documented to be system-wide), so perhaps the correct solution would be to move it to `ApplicationHandler`?

This sounds good to me.